src/http.rs

```diff
@@ -123,6 +124,17 @@ where
                 })
                 .context(CallRequestSnafu)?;
 
+            let (parts, body) = response.into_parts();
+            let mut buf: Vec<u8> = Vec::new();
+            let body = body.map_data(move |data| {
+                buf.extend(data);
+                if buf.len() > 20 {
+                    println!("Data buf has reached len 20: {:?}", std::str::from_utf8(&buf));
```

Comment:
Currently I am printing to stdout, I would like to not do this, what steps will I need to take to implement viewing within the section `body` of the below line:

```
Nov 12 20:03:22.143 DEBUG http: vector::http: Sending HTTP request. uri=http://localhost:8080/_bulk?timeout=60s method=POST version=HTTP/1.1 headers={"content-type": "application/x-ndjson", "user-agent": "Vector/0.11.0 (g9ba2c93 x86_64-unknown-linux-gnu 2020-11-12)"} body=[245 bytes]
```
I am aware of the `std::fmt::Display` for `FormatBody` implementation located in `src/internal_events/http_client.rs` and believe that's where I will make the changes I'm interested in. However I struggle to see how I will pass the data from the vec and have that be used within that `std::fmt::Display` scope.

Reply:
feedback from Bruce: 
1. Create an internal event, that takes as parameter the reference of the buffer and then emit that internal event
2. This will be in the `src/internal_events/http_client.rs`
3. Within that internal event will use the macro debug, which will takes named parameters which will have message = incoming data and data = something

Reply:
There is a logic bug 

Make the buffer an option, and once reaches the size you take it and then it will become a None
Take is a function in Option and sets the internal data to empty. 
Look into `Option.take`

Reply:
The most recent commits seem to implement the desired behavior